src/mattack_actors.cpp

```diff
@@ -384,26 +387,28 @@ void gun_actor::shoot( monster &z, Creature &target ) const
     tmp.dex_cur = fake_dex;
     tmp.int_cur = fake_int;
     tmp.per_cur = fake_per;
+    tmp.attitude = z.friendly ? NPCATT_DEFEND : NPCATT_KILL;
 
-    if( z.friendly != 0 ) {
-        tmp.attitude = NPCATT_DEFEND;
-    } else {
-        tmp.attitude = NPCATT_KILL;
+    if( fake_skills.empty() ) {
+        tmp.skillLevel( skill_id( "gun" ) ).level( 4 );
+        tmp.skillLevel( gun.gun_skill() ).level( 8 );
     }
-
     for( const auto &pr : fake_skills ) {
         tmp.skillLevel( pr.first ).level( pr.second );
     }
 
-    tmp.weapon = item( gun_type ).ammo_set( ammo_type, z.ammo[ ammo_type ] );
+    tmp.weapon = gun;
+    tmp.worn.push_back( item( "fake_UPS", calendar::turn, 1000 ) );
 
     const auto distance = rl_dist( z.pos(), target.pos() );
-    int burst_size = std::min( burst_limit, tmp.weapon.burst_size() );
-    if( distance > range_no_burst ) {
-        burst_size = 1;
+
+    const int burst = std::min( distance <= range_no_burst ? burst_limit : 1, tmp.weapon.burst_size() );
+
+    if( g->u.sees( z ) ) {
+        add_msg( m_warning, _( description.c_str() ), z.name().c_str(), tmp.weapon.gun_type().c_str() );
```

Comment:
The result of `item::gun_type` needs to be translated when you're printing it. See implementation of that function (it returns an untranslated string) and its usage in `item::gunmod_compatible` (where it is translated for printing).